### DISCUSSION TITLE
leader key

### DISCUSSION THREAD

**Discussion by danielb2** [zellij-org/zellij#2982] [Ideas]
playing around with zellij my biggest concern is how many key-bindings it's consuming given the lack of a leader key (such as ctrl+a for screen and ctrl+b for tmux).

For example, in vim, my ctrl+p and ctrl+t are mapped already. so how can I use them in vim now?    is there a way to configure zellij to have a leader key? if not, maybe that's a feature worth considering.

This is stopping me from migrating. That those bindings are already assigned are an example. The basic idea is I don't want conflicting keybindings with other terminal apps I may be using. What's not conflicting in Vim may have a conflicting somewhere else, so please don't just respond with changing my keybindings in vim.

FYi, I tried removing

```
shared_except "pane" "locked" {
    bind "Ctrl p" { SwitchToMode "Pane"; }
}                          
```

in config.kdl and for ctrl t as well, but this doesn't seem to have worked

**Comment by danielb2** [comment]
seems that adding

      keybinds {
         normal {
            unbind "Ctrl p"
            unbind "Ctrl t"
         }
      }

did what I wanted

**Comment by Lockszmith-GH** [comment]
Not a leader key, but the situation is addressed in a couple of ways:
* From the FAQ: [Zellij overrides certain key combinations that I use for other apps, what can I do?](https://zellij.dev/documentation/faq.html?highlight=locked#zellij-overrides-certain-key-combinations-that-i-use-for-other-apps-what-can-i-do)
* And mentioned in the [top line of the default.kdl](https://github.com/zellij-org/zellij/blob/main/zellij-utils/assets/config/default.kdl#L1): using `keybinds clear-defaults=true`

There were attempts at leader keys as well, I think you can still acheive this by programing keys to perform an action and drop to command mode.

**Comment by danielb2** [comment]
somehow I missed the top comment there. that does it. thanks

**Comment by merikan** [comment]
@danielb2 
When I switched from Tmux to Zellij a year ago I had the same problem. I therefore removed all Ctrl-bindings from Normal-mode and put them under Tmux-mode instead. In Tmux-mode I have some direct bindings for what I use most, split pane, create tab etc., just like I had in Tmux. If I want to go to another mode e.g. Session-mode, I go via Tmux mode.

Have a look at my [Readme](https://github.com/merikan/.dotfiles/blob/main/config/zellij/readme.md) and you can find my config here https://github.com/merikan/.dotfiles/blob/main/config/zellij/config.kdl

**Reply by danielb2** [reply]
sweet. this is useful. 

thanks. I think you should do `keybinds clear-defaults=true` too. probably don't need it for the other groups.

I also added the shortcuts for changing tabs as I saw they were missing. I noticed your rename tab is not the `,` for tmux which I switched it to. 

Out of curiousity, what are you liking more about zellij than tmux?

**Reply by Lockszmith-GH** [reply]
@imsnif would it make sense to curate some 'alternative configs' in the code base/documentation site?
Like this config.kdl shared by merikan?

I'm willing to deal with the PR, just wanted to know where this would fit.

**Reply by merikan** [reply]
@danielb2 
I am doing `clear-defaults true` for `tmux` and `normal` mode, the rest is kept as before.

Regarding mappings: I have used Tmux for many years (15+) and with a Swedish keyboard, I am now using US, the default mappings didn't make any sense. It's now in my muscle memory but it's never too late to change. ;-)  To be honest, I never really understood Tmux default bindings.

One thing is the plugin system with WASM, it is so cool. Another thing is that when you zoom a pane you can then switch between panes in zoom mode, in Tmux when trying to switch in zoom mode you will fall out from zoom mode.  Of course, many little things need to be fixed before Zellij is equivalent to Tmux, but it works for my needs at the moment.
Another thing, besides technology, is that I perceive the founder as a good and nice person, and that influences me to want to use and recommend a software.

@Lockszmith-GH
I agree with you, It should be in the documentation. I see that question pop up occasionally, and it's an unnecessary blocker/concern for many who want to use Zellij.

**Reply by imsnif** [reply]
Friends, I'm sorry - whatever goes into this repository I will have to maintain, and I'm not keeping up as is. I'm working on doing just the opposite (offloading functionality into plugins, allowing people to develop cool plugins that they can easily share with others, working on various integrations eg. through layouts to allow people to express their creativity, etc.)

I will warmly encourage you to host these configs and whatever other user content you'd like on your own. The more this happens, the healthier the ecosystem becomes.

**Reply by Lockszmith-GH** [reply]
Fair enough. Would a community resource page in the documentation work?
Placing a link to other repos/doc sites?

**Reply by imsnif** [reply]
...which I would have to curate, I guess? :)

How about you create an `awesome-zellij` list?

**Reply by Lockszmith-GH** [reply]
That's what I was thinking about - something of that sort. I've never done one of those, but I'll definitly try :)

**Reply by merikan** [reply]
@imsnif  @Lockszmith-GH  
I agree, whatever goes into this repository someone has to maintain. An `awesome-zellij` list is definitely a good thing that can promote cool plugins, themes, configs etc. Another option could be to use the wiki in the repo where sections could be community maintained. This is done in many other repos such as [fzf](https://github.com/junegunn/fzf/wiki) and [spaceship-prompt](https://github.com/spaceship-prompt/spaceship-prompt/wiki/External-sections). 

If I understand it correctly @imsnif , you are the sole maintainer and we should absolutely not put more burden on you, so let us start with an `awesome-zellij` repo 👍 Go for it @Lockszmith-GH
